File: hrd_exam1/src/main/webapp/join.js
console.log('join js 열림')



function join(){
	console.log('join 버튼')
	
	//  유효성 검사
	if(document.joinform.custname.value==""){
		alert('회원 성명을 입력해주세요')
		document.joinform.custname.focus(); // 커서이동
		return false;
	}
	if(document.joinform.phone.value==""){
		alert('회원 전화를 입력해주세요')
		document.joinform.phone.focus(); // 커서이동
		return false;
	}
	if(document.joinform.address.value==""){
		alert('회원 주소를 입력해주세요')
		document.joinform.address.focus(); // 커서이동
		return false;
	}
	if(document.joinform.joindate.value==""){
		alert('가입 일자를 입력해주세요')
		document.joinform.joindate.focus(); // 커서이동
		return false;
	}
	if(document.joinform.grade.value==""){
		alert('고객등급를 입력해주세요')
		document.joinform.cigradety.focus(); // 커서이동
		return false;
	}
	if(document.joinform.city.value==""){
		alert('도시코드를 입력해주세요')
		document.joinform.city.focus(); // 커서이동
		return false;
	}
	
	
	document.joinform.submit();
	alert('회원가입이 완료되었습니다.')
	
	
}

File: hrd_exam1/src/main/java/controller/Join.java
package controller;

import java.io.IOException;
import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import Dto.MemberDto;


@WebServlet("/join")
public class Join extends HttpServlet {
	private static final long serialVersionUID = 1L;
       
    
    public Join() {super();}

	
	protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
		request.setCharacterEncoding("UTF-8");
		
		int custno = Integer.parseInt(request.getParameter("custno"));
		String custname = request.getParameter("custname");
		String phone = request.getParameter("phone");
		String address = request.getParameter("address");
		String joindate = request.getParameter("joindate");
		String grade = request.getParameter("grade");
		String city = request.getParameter("city");
		
		MemberDto dto = new MemberDto(custno, custname, phone, address, joindate, grade, city);

		Dao dao = new Dao();
		dao.update(dto);
		response.sendRedirect("/hrd_exam1/update.jsp?custno="+custno);
		
		
	}

	protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
		
		request.setCharacterEncoding("UTF-8");
		int custno = Integer.parseInt(request.getParameter("custno"));
		String custname = request.getParameter("custname");
		String phone = request.getParameter("phone");
		String address = request.getParameter("address");
		String joindate = request.getParameter("joindate");
		String grade = request.getParameter("grade");
		String city = request.getParameter("city");
		
		MemberDto dto = new MemberDto(custno, custname, phone, address, joindate, grade, city);
		
		Dao dao = new Dao();
		boolean result = dao.join(dto);
		
		if(result) {
			response.setCharacterEncoding("UTF-8");
			response.sendRedirect("/hrd_exam1/join.jsp");
		}
		
		
	}

}
